### PR TITLE
Add queue scores by user id

### PR DESCRIPTION
**Pull request by nanaya**
To be used by user reindex process (LIO userIndex).

### CODE REVIEW

**Review by notbakaneko — CHANGES_REQUESTED** [review]
Having to use `--user=1 --all` seems unintuitive especially if `user` is being ignored for `ids`. Then there's `--user=1 --from=100` which is actually `user = 1 or id > 100`...

**Comment by nanaya** [comment]
I'm fixing this mixing parameter nonsense ಠ_ಠ

File: app/Console/Commands/EsIndexScoresQueue.php [hunk 1]
```diff
@@ -104,11 +90,66 @@ public function handle()
...
+        } else {
+            $this->schemas = [$schema];
+        }
+    }
+
     private function parseOptionIds(): Set
     {
         $ret = new Set();
+        $ids = $this->option('ids');
+        $ids = is_array($ids) ? $ids : explode(',', get_string($ids) ?? '');
```
Comment (notbakaneko) [inline app/Console/Commands/EsIndexScoresQueue.php:150]: seems strange that `$ids` allows array without `--ids=*` as option signature 🤔  (and weirder that laravel doesn't enforce `*` when passed as direct argument 🤔 )

File: tests/Commands/EsIndexScoresQueueTest.php [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+            array_merge(is_callable($params) ? $params() : $params, [
+                '--no-interaction' => true,
+                '--schema' => static::SCHEMA,
+            ]),
+        );
+    }
+
+    public function dataProviderForTestParameterValidity(): array
+    {
+        return [
```
Comment (notbakaneko) [inline tests/Commands/EsIndexScoresQueueTest.php:64]: seems missing `--user=0` case

**Review by notbakaneko — CHANGES_REQUESTED** [review]

File: app/Console/Commands/EsIndexScoresQueue.php (hunk 1 above)
Comment (nanaya) [reply app/Console/Commands/EsIndexScoresQueue.php:150]: I don't think supporting that for cli is a good idea... The resulting option would then always be in an array and if we're going to still support comma separated value it also means additional parsing inside the array.

Laravel doesn't seem to do anything with the param values when passed using `Artisan::call`.

**Review by notbakaneko — APPROVED** [review]

**Pull request merged**